internal/cmd/api/api.go

```diff
@@ -0,0 +1,332 @@
+
+		HEADERS
+
+			Use the --header/-H flag to specify headers manually. Default headers are already included such as "User-Agent",
+			"Content-Type", "Accept" and "Authorization".
+
+		BODY
+
+			If a body is required for the endpoint you're targeting, the --field/-F flag allows you to specify values at specific
+			paths of a JSON document. Repeat the flag to set multiple values. All bodies sent are JSON objects, it isn't possible
+			to send non JSON content, or JSON entities of non-object type as the root of the body.
+
+			If you wish to pass the content of a file as body, use the --input/-I flag. If the body should be read from stdin,
+			use --input="-".
+
+			Fields specified with --field/-F will be updated in the content of --input/-I if the content is JSON. If not, an error
+			will be returned.
+		`),
+		Example: heredoc.Docf(`
+		# get the current user
+
+		$ pscale api user
+
+		# list an org's databases
+
+		$ pscale api organizations/{org}/databases
+
+		# create a database
+
+		$ pscale api organizations/{org}/databases -F 'name="my-database"'
+
+		# get a database
+
+		$ pscale api organizations/{org}/databases/my-database
+
+		# delete a database
+
+		$ pscale api -X DELETE organizations/{org}/databases/my-database
+
+		# add a note on a database from the content of a file
+
+		$ pscale api -X PATCH organizations/{org}/databases/{db} -F 'notes=@mynotes.txt'
+
+		# create a database branch from a file
+
+		$ pscale api organizations/{org}/databases/{db}/branches --input=spec.json
+
+		# create a database branch from stdin, override the name
+
+		$ pscale api organizations/{org}/databases/{db}/branches --input=- -F 'name="my-name"'
+
+		`),
+		Args:              cobra.ExactArgs(1),
+		PersistentPreRunE: cmdutil.CheckAuthentication(ch.Config),
+	}
+
+	cmd.PersistentFlags().StringVar(&ch.Config.Organization, "org", ch.Config.Organization, "The organization for the current user")
+	cmd.PersistentFlags().StringVar(&ch.Config.Database, "database", ch.Config.Database, "The database this project is using")
+	cmd.PersistentFlags().StringVar(&ch.Config.Branch, "branch", ch.Config.Branch, "The branch this project is using")
+
+	opts := &ApiOpts{}
+	cmd.Flags().StringVarP(&opts.Method, "method", "X", "GET", "HTTP method to use for the request")
+	cmd.Flags().StringArrayVarP(&opts.Query, "query", "Q", nil, "query to append to the URL path, in `key=value` format")
+	cmd.Flags().StringArrayVarP(&opts.Header, "header", "H", nil, "HTTP headers to add to the request")
+	cmd.Flags().StringArrayVarP(&opts.Field, "field", "F", nil, "HTTP body to send with the request, in `key=value` format where `value` is a JSON entity, unless `value` starts with a `@` in which case the string after `@` represents a file that will be read. Nested types are represented as `root.depth1.depth2=value``")
+	cmd.Flags().StringVarP(&opts.Input, "input", "I", "", "HTTP body to send with the request, as a file that will be read and then sent.")
```

Comment:
I disagree, there are cases where passing via stdin is hard or impossible (crappy CI systems with crappy shell command interpretation). Also passing arguments via flags (`-F`) or via a file (`-I`) handles different use cases. The `-F` is useful in cases like CI where you want to set fields from a templating engine,  and `-I` is useful when you want to use a file that's checked in a repo or some such. Allowing both together makes it possible to customize.

Handling `stdin` input by detection of what's on `fd/0` is nice for interactive commands or pure commands, but since `pscale` is mostly for human, but `pscale api` is meant to be a safe-haven for scripting, I think it's best if we make things explicit (`-I="-"`) instead of magic. Because magic can break between releases, and `pscale api` is meant to be the stable section of `pscale`.

Reply:
Makes sense, thanks for the clarification.